### PR TITLE
docs(logger): add logEventIfEnabled() docs

### LINKED ISSUE #1904: Feature request: Support Logging Incoming Event Without Use of Middy or Class
### Use case

I want the incoming log event to be automatically logged for me when `POWERTOOLS_LOGGER_LOG_EVENT` is set to true and I'm not using middy or a handler class. Middy and handler class usage is demonstrated in docs [here](https://docs.powertools.aws.dev/lambda/typescript/latest/core/logger/#log-incoming-event).

### Solution/User Experience

Conditionally log incoming event based on `POWERTOOLS_LOGGER_LOG_EVENT` within [Logger.addContext](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/logger/src/Logger.ts#L189). I know this may seem strange to put "log incoming event logic" into the `addContext` method, but that's how it's implemented with middy (`injectLambdaContext`) and class decorator (`logger.injectLambdaContext`).

With this solution, the below function would log incoming event. Currently it does not.
```ts
import { Logger } from "@aws-lambda-powertools/logger";
process.env.POWERTOOLS_LOGGER_LOG_EVENT = "true";
const logger = new Logger();
export function handler(event, context) {
  logger.addContext(context);
  // do work
}
```

### Alternative solutions

_No response_

### Acknowledgment

- [X] This feature request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR adds to both the documentation and API docs info about the previously undocumented `logEventIfEnabled()` method in Logger.

This method allows customers who are not using Middy.js middleware nor class method decorators to optionally log the incoming event when the `POWERTOOLS_LOGGER_LOG_EVENT` environment variable is set to `true`.

The updated section in the Logger docs looks like this:
Screenshot 2024-08-14 at 12 14 25

The newly added code example, and its tooltip, look like this:
Screenshot 2024-08-14 at 12 14 30

Similar tooltips have been added to the Middy.js middleware and class method decorator examples:
Screenshot 2024-08-14 at 12 14 42

Finally, I have also added an example event and output to show what it's going to be logged:
Screenshot 2024-08-14 at 12 14 49

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #1904

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: examples/snippets/logger/logEventDecorator.ts [hunk 1]
```diff
@@ -4,12 +4,11 @@ import { Logger } from '@aws-lambda-powertools/logger';
 const logger = new Logger();
 
 class Lambda implements LambdaInterface {
-  // Set the log event flag to true
-  @logger.injectLambdaContext({ logEvent: true })
+  @logger.injectLambdaContext({ logEvent: true }) // (1)
   public async handler(_event: unknown, _context: unknown): Promise<void> {
-    logger.info('This is an INFO log with some context');
+    // ... your lambda handler
```
Comment (leandrodamascena) [inline examples/snippets/logger/logEventDecorator.ts:9]:
```suggestion
    // ... your logic here
```

File: examples/snippets/logger/logEventManual.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+process.env.POWERTOOLS_LOGGER_LOG_EVENT = 'true';
+
+import { Logger } from '@aws-lambda-powertools/logger';
+
+const logger = new Logger();
+
+export const handler = async (event: unknown) => {
+  logger.logEventIfEnabled(event); // (1)
+  // ... your handler code
```
Comment (leandrodamascena) [inline examples/snippets/logger/logEventManual.ts:9]:
```suggestion
  // ... your logic here
```

File: examples/snippets/logger/logEventMiddy.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { Logger } from '@aws-lambda-powertools/logger';
+import { injectLambdaContext } from '@aws-lambda-powertools/logger/middleware';
+import middy from '@middy/core';
+
+const logger = new Logger();
+
+export const handler = middy(async () => {
+  // ... your lambda handler
```
Comment (leandrodamascena) [inline examples/snippets/logger/logEventMiddy.ts:8]:
```suggestion
  // ... your logic here
```

**Review by leandrodamascena — DISMISSED** [review]
Just a small text correction, but not mandatory. Feel free to not accept and merge without them.

**Comment by dreamorosi** [comment]
> Just a small text correction, but not mandatory. Feel free to not accept and merge without them.

No, they are good suggestions - accepted all of them.

**Review by leandrodamascena — APPROVED** [review]
APPROVED!

**Pull request merged**